### ISSUE TITLE
Can't use `*` in regex

### ISSUE THREAD

**Issue by ingydotnet** [koalaman/shellcheck#1367]
#### For bugs
- Rule Id 2049
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
echo 'x="foo*"; [[ $x =~ (o*) ]]' | shellcheck -s bash -
```

#### Here's what shellcheck currently says:
```
In - line 1:
x="foo*"; [[ $x =~ (o*) ]]
                   ^-- SC2049: =~ is for regex. Use == for globs.
```

#### Here's what I wanted or expected to see:

Should be fine to use `*` in a regex. `+` and `?` both work here.

**Comment by koalaman** [comment]
The problem with this regex is that it always matches. ShellCheck thought you may have intended to use `o.*` instead (or to treat it as a glob).

This was compounded by some parsing bugs that dropped the `()` , and a missing wiki page, which are now fixed. `(o*)` will not be producing this warning the way just `o*` does.
